kv/kvserver/asim/tests: TestDataDriven failed

kv/kvserver/asim/tests.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/21064504?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyCockroachExtraStress/21064504?buildTab=artifacts#/) on refs/heads/master @ [3f2e2d64d62837f5dee10b1988a2cfedb013091e](https://github.com/cockroachdb/cockroach/commits/3f2e2d64d62837f5dee10b1988a2cfedb013091e):


```
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2595/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma/sma_vs_mma_rebalance_speed.txt:78: still running after 1m20.047384083s
    datadriven_simulation_test.go:213: 
          /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2595/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/asim/tests/tests_test_/tests_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma/sma_vs_mma_rebalance_speed.txt:78:
          eval
        output didn't match expected:
          @@ -1,37 +1,40 @@
          +sma-count:
           cpu#1: last:  [s1=1650518518, s2=1725373840, s3=1625615244] (stddev=42394011.32, mean=1667169200.67, sum=5001507602)
           cpu#1: thrash_pct: [s1=4%, s2=5%, s3=5%]  (sum=14%)
           cpu_util#1: last:  [s1=0.21, s2=0.22, s3=0.20] (stddev=0.01, mean=0.21, sum=1)
           cpu_util#1: thrash_pct: [s1=4%, s2=5%, s3=5%]  (sum=14%)
           leases#1: first: [s1=200, s2=0, s3=0] (stddev=94.28, mean=66.67, sum=200)
           leases#1: last:  [s1=66, s2=69, s3=65] (stddev=1.70, mean=66.67, sum=200)
           leases#1: thrash_pct: [s1=0%, s2=0%, s3=0%]  (sum=0%)
           replicas#1: first: [s1=200, s2=200, s3=200] (stddev=0.00, mean=200.00, sum=600)
           replicas#1: last:  [s1=200, s2=200, s3=200] (stddev=0.00, mean=200.00, sum=600)
           replicas#1: thrash_pct: [s1=0%, s2=0%, s3=0%]  (sum=0%)
          -artifacts[sma-count]: d29c210445a690ce
          +hash: d29c210445a690ce
           ==========================
          +mma-only:
           cpu#1: last:  [s1=1820976543, s2=1649994712, s3=1524369164] (stddev=121560456.89, mean=1665113473.00, sum=4995340419)
           cpu#1: thrash_pct: [s1=2%, s2=11%, s3=11%]  (sum=25%)
           cpu_util#1: last:  [s1=0.23, s2=0.21, s3=0.19] (stddev=0.02, mean=0.21, sum=1)
           cpu_util#1: thrash_pct: [s1=2%, s2=11%, s3=11%]  (sum=25%)
           leases#1: first: [s1=200, s2=0, s3=0] (stddev=94.28, mean=66.67, sum=200)
           leases#1: last:  [s1=73, s2=66, s3=61] (stddev=4.92, mean=66.67, sum=200)
           leases#1: thrash_pct: [s1=0%, s2=0%, s3=0%]  (sum=0%)
           replicas#1: first: [s1=200, s2=200, s3=200] (stddev=0.00, mean=200.00, sum=600)
           replicas#1: last:  [s1=200, s2=200, s3=200] (stddev=0.00, mean=200.00, sum=600)
           replicas#1: thrash_pct: [s1=0%, s2=0%, s3=0%]  (sum=0%)
          -artifacts[mma-only]: 239cb1f32d4a2e3f
          +hash: 239cb1f32d4a2e3f
           ==========================
          +mma-count:
           cpu#1: last:  [s1=1776335802, s2=1675272645, s3=1549901392] (stddev=92618845.46, mean=1667169946.33, sum=5001509839)
           cpu#1: thrash_pct: [s1=4%, s2=6%, s3=5%]  (sum=15%)
           cpu_util#1: last:  [s1=0.22, s2=0.21, s3=0.19] (stddev=0.01, mean=0.21, sum=1)
           cpu_util#1: thrash_pct: [s1=4%, s2=6%, s3=5%]  (sum=15%)
           leases#1: first: [s1=200, s2=0, s3=0] (stddev=94.28, mean=66.67, sum=200)
           leases#1: last:  [s1=71, s2=67, s3=62] (stddev=3.68, mean=66.67, sum=200)
           leases#1: thrash_pct: [s1=0%, s2=0%, s3=0%]  (sum=0%)
           replicas#1: first: [s1=200, s2=200, s3=200] (stddev=0.00, mean=200.00, sum=600)
           replicas#1: last:  [s1=200, s2=200, s3=200] (stddev=0.00, mean=200.00, sum=600)
           replicas#1: thrash_pct: [s1=0%, s2=0%, s3=0%]  (sum=0%)
          -artifacts[mma-count]: c3c6de6cbcc1f69f
          +hash: c3c6de6cbcc1f69f
           ==========================
           
--- FAIL: TestDataDriven/mode=mma/sma_vs_mma_rebalance_speed (85.74s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-59397

--- a/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma/constraint_satisfaction_increase_disk_util.txt
+++ b/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma/constraint_satisfaction_increase_disk_util.txt
@@ -42,6 +42,7 @@ assertion type=conformance under=0 over=0 unavailable=0 violating=0
 
 eval duration=10m samples=1 seed=42 cfgs=(sma-count,mma-count) metrics=(cpu,cpu_util,leases,replicas,disk_fraction_used)
 ----
+sma-count:
 disk_fraction_used#1: first: [s1=0.73, s2=0.73, s3=0.37, s4=0.37, s5=0.18, s6=0.18, s7=0.18] (stddev=0.23, mean=0.39, sum=3)
 disk_fraction_used#1: last:  [s1=0.92, s2=0.92, s3=0.37, s4=0.37, s5=0.06, s6=0.06, s7=0.06] (stddev=0.35, mean=0.39, sum=3)
 disk_fraction_used#1: thrash_pct: [s1=0%, s2=0%, s3=0%, s4=0%, s5=0%, s6=0%, s7=0%]  (sum=0%)
@@ -51,8 +52,9 @@ leases#1: thrash_pct: [s1=20%, s2=0%, s3=0%, s4=0%, s5=0%, s6=0%, s7=0%]  (sum=2
 replicas#1: first: [s1=60, s2=60, s3=30, s4=30, s5=15, s6=15, s7=15] (stddev=18.68, mean=32.14, sum=225)
 replicas#1: last:  [s1=75, s2=75, s3=30, s4=30, s5=5, s6=5, s7=5] (stddev=29.01, mean=32.14, sum=225)
 replicas#1: thrash_pct: [s1=0%, s2=0%, s3=0%, s4=0%, s5=0%, s6=0%, s7=0%]  (sum=0%)
-artifacts[sma-count]: 9e69db6ac95253c2
+hash: 9e69db6ac95253c2
 ==========================
+mma-count:
 disk_fraction_used#1: first: [s1=0.73, s2=0.73, s3=0.37, s4=0.37, s5=0.18, s6=0.18, s7=0.18] (stddev=0.23, mean=0.39, sum=3)
 disk_fraction_used#1: last:  [s1=0.92, s2=0.92, s3=0.37, s4=0.37, s5=0.06, s6=0.06, s7=0.06] (stddev=0.35, mean=0.39, sum=3)
 disk_fraction_used#1: thrash_pct: [s1=0%, s2=0%, s3=0%, s4=0%, s5=0%, s6=0%, s7=0%]  (sum=0%)
@@ -62,5 +64,5 @@ leases#1: thrash_pct: [s1=20%, s2=0%, s3=0%, s4=0%, s5=0%, s6=0%, s7=0%]  (sum=2
 replicas#1: first: [s1=60, s2=60, s3=30, s4=30, s5=15, s6=15, s7=15] (stddev=18.68, mean=32.14, sum=225)
 replicas#1: last:  [s1=75, s2=75, s3=30, s4=30, s5=5, s6=5, s7=5] (stddev=29.01, mean=32.14, sum=225)
 replicas#1: thrash_pct: [s1=0%, s2=0%, s3=0%, s4=0%, s5=0%, s6=0%, s7=0%]  (sum=0%)
-artifacts[mma-count]: 9e69db6ac95253c2
+hash: 9e69db6ac95253c2
 ==========================

--- a/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma/constraint_satisfaction_no_rebalance_count.txt
+++ b/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma/constraint_satisfaction_no_rebalance_count.txt
@@ -35,6 +35,7 @@ setting split_queue_enabled=false
 
 eval duration=5m samples=1 seed=42 cfgs=(sma-count) metrics=(leases,replicas,disk_fraction_used)
 ----
+sma-count:
 disk_fraction_used#1: first: [s1=0.61, s2=0.61] (stddev=0.00, mean=0.61, sum=1)
 disk_fraction_used#1: last:  [s1=0.33, s2=0.89] (stddev=0.28, mean=0.61, sum=1)
 disk_fraction_used#1: thrash_pct: [s1=0%, s2=0%]  (sum=0%)
@@ -44,7 +45,7 @@ leases#1: thrash_pct: [s1=0%, s2=0%]  (sum=0%)
 replicas#1: first: [s1=50, s2=3] (stddev=23.50, mean=26.50, sum=53)
 replicas#1: last:  [s1=27, s2=26] (stddev=0.50, mean=26.50, sum=53)
 replicas#1: thrash_pct: [s1=0%, s2=0%]  (sum=0%)
-artifacts[sma-count]: 48dbff430f9a21be
+hash: 48dbff430f9a21be
 ==========================
 
 assertion type=balance ticks=5 stat=disk_fraction_used upper_bound=1.1
@@ -60,6 +61,7 @@ assertion type=conformance under=0 over=0 unavailable=0 violating=0
 
 eval duration=5m samples=1 seed=42 cfgs=(mma-count) metrics=(leases,replicas,disk_fraction_used)
 ----
+mma-count:
 disk_fraction_used#1: first: [s1=0.61, s2=0.61] (stddev=0.00, mean=0.61, sum=1)
 disk_fraction_used#1: last:  [s1=0.57, s2=0.65] (stddev=0.04, mean=0.61, sum=1)
 disk_fraction_used#1: thrash_pct: [s1=0%, s2=0%]  (sum=0%)
@@ -69,5 +71,5 @@ leases#1: thrash_pct: [s1=0%, s2=0%]  (sum=0%)
 replicas#1: first: [s1=50, s2=3] (stddev=23.50, mean=26.50, sum=53)
 replicas#1: last:  [s1=47, s2=6] (stddev=20.50, mean=26.50, sum=53)
 replicas#1: thrash_pct: [s1=0%, s2=0%]  (sum=0%)
-artifacts[mma-count]: d2a2823701a6f92a
+hash: d2a2823701a6f92a
 ==========================

--- a/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma/constraint_satisfaction_no_rebalance_disk.txt
+++ b/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma/constraint_satisfaction_no_rebalance_disk.txt
@@ -40,6 +40,7 @@ assertion type=conformance under=0 over=0 unavailable=0 violating=0
 
 eval duration=3m samples=1 seed=42 cfgs=(sma-count,mma-count) metrics=(cpu,cpu_util,leases,replicas,disk_fraction_used)
 ----
+sma-count:
 disk_fraction_used#1: first: [s1=0.73, s2=0.73, s3=0.49, s4=0.49, s5=0.49, s6=0.73] (stddev=0.12, mean=0.61, sum=4)
 disk_fraction_used#1: last:  [s1=0.73, s2=0.73, s3=0.49, s4=0.49, s5=0.49, s6=0.73] (stddev=0.12, mean=0.61, sum=4)
 disk_fraction_used#1: thrash_pct: [s1=0%, s2=0%, s3=0%, s4=0%, s5=0%, s6=0%]  (sum=0%)
@@ -49,8 +50,9 @@ leases#1: thrash_pct: [s1=0%, s2=0%, s3=0%, s4=0%, s5=0%, s6=0%]  (sum=0%)
 replicas#1: first: [s1=12, s2=12, s3=8, s4=8, s5=8, s6=12] (stddev=2.00, mean=10.00, sum=60)
 replicas#1: last:  [s1=12, s2=12, s3=8, s4=8, s5=8, s6=12] (stddev=2.00, mean=10.00, sum=60)
 replicas#1: thrash_pct: [s1=0%, s2=0%, s3=0%, s4=0%, s5=0%, s6=0%]  (sum=0%)
-artifacts[sma-count]: 8171cff9be3d23
+hash: 8171cff9be3d23
 ==========================
+mma-count:
 disk_fraction_used#1: first: [s1=0.73, s2=0.73, s3=0.49, s4=0.49, s5=0.49, s6=0.73] (stddev=0.12, mean=0.61, sum=4)
 disk_fraction_used#1: last:  [s1=0.73, s2=0.73, s3=0.49, s4=0.49, s5=0.49, s6=0.73] (stddev=0.12, mean=0.61, sum=4)
 disk_fraction_used#1: thrash_pct: [s1=0%, s2=0%, s3=0%, s4=0%, s5=0%, s6=0%]  (sum=0%)
@@ -60,5 +62,5 @@ leases#1: thrash_pct: [s1=0%, s2=0%, s3=0%, s4=0%, s5=0%, s6=0%]  (sum=0%)
 replicas#1: first: [s1=12, s2=12, s3=8, s4=8, s5=8, s6=12] (stddev=2.00, mean=10.00, sum=60)
 replicas#1: last:  [s1=12, s2=12, s3=8, s4=8, s5=8, s6=12] (stddev=2.00, mean=10.00, sum=60)
 replicas#1: thrash_pct: [s1=0%, s2=0%, s3=0%, s4=0%, s5=0%, s6=0%]  (sum=0%)
-artifacts[mma-count]: 8171cff9be3d23
+hash: 8171cff9be3d23
 ==========================

--- a/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma/sma_vs_mma_rebalance_speed.txt
+++ b/pkg/kv/kvserver/asim/tests/testdata/non_rand/mma/sma_vs_mma_rebalance_speed.txt
@@ -77,6 +77,7 @@ asserting: max_{stores}(cpu)/mean_{stores}(cpu) ≤ 1.15 at each of last 6 ticks
 # takes ~5min.
 eval duration=32m samples=1 seed=42 cfgs=(sma-count,mma-only,mma-count) metrics=(cpu,cpu_util,leases,replicas)
 ----
+sma-count:
 cpu#1: last:  [s1=1650518518, s2=1725373840, s3=1625615244] (stddev=42394011.32, mean=1667169200.67, sum=5001507602)
 cpu#1: thrash_pct: [s1=4%, s2=5%, s3=5%]  (sum=14%)
 cpu_util#1: last:  [s1=0.21, s2=0.22, s3=0.20] (stddev=0.01, mean=0.21, sum=1)
@@ -87,8 +88,9 @@ leases#1: thrash_pct: [s1=0%, s2=0%, s3=0%]  (sum=0%)
 replicas#1: first: [s1=200, s2=200, s3=200] (stddev=0.00, mean=200.00, sum=600)
 replicas#1: last:  [s1=200, s2=200, s3=200] (stddev=0.00, mean=200.00, sum=600)
 replicas#1: thrash_pct: [s1=0%, s2=0%, s3=0%]  (sum=0%)
-artifacts[sma-count]: d29c210445a690ce
+hash: d29c210445a690ce
 ==========================
+mma-only:
 cpu#1: last:  [s1=1820976543, s2=1649994712, s3=1524369164] (stddev=121560456.89, mean=1665113473.00, sum=4995340419)
 cpu#1: thrash_pct: [s1=2%, s2=11%, s3=11%]  (sum=25%)
 cpu_util#1: last:  [s1=0.23, s2=0.21, s3=0.19] (stddev=0.02, mean=0.21, sum=1)
@@ -99,8 +101,9 @@ leases#1: thrash_pct: [s1=0%, s2=0%, s3=0%]  (sum=0%)
 replicas#1: first: [s1=200, s2=200, s3=200] (stddev=0.00, mean=200.00, sum=600)
 replicas#1: last:  [s1=200, s2=200, s3=200] (stddev=0.00, mean=200.00, sum=600)
 replicas#1: thrash_pct: [s1=0%, s2=0%, s3=0%]  (sum=0%)
-artifacts[mma-only]: 239cb1f32d4a2e3f
+hash: 239cb1f32d4a2e3f
 ==========================
+mma-count:
 cpu#1: last:  [s1=1776335802, s2=1675272645, s3=1549901392] (stddev=92618845.46, mean=1667169946.33, sum=5001509839)
 cpu#1: thrash_pct: [s1=4%, s2=6%, s3=5%]  (sum=15%)
 cpu_util#1: last:  [s1=0.22, s2=0.21, s3=0.19] (stddev=0.01, mean=0.21, sum=1)
@@ -111,5 +114,5 @@ leases#1: thrash_pct: [s1=0%, s2=0%, s3=0%]  (sum=0%)
 replicas#1: first: [s1=200, s2=200, s3=200] (stddev=0.00, mean=200.00, sum=600)
 replicas#1: last:  [s1=200, s2=200, s3=200] (stddev=0.00, mean=200.00, sum=600)
 replicas#1: thrash_pct: [s1=0%, s2=0%, s3=0%]  (sum=0%)
-artifacts[mma-count]: c3c6de6cbcc1f69f
+hash: c3c6de6cbcc1f69f
 ==========================

